packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
+  /**
+   * The value of the option.
+   */
+  readonly value: string | number;
```

Comment:
Should probably be a generic so types could be inferred by the ones passed in rather than needing to coerce it.